Service/Worker/AbstractWorker.php

```diff
@@ -56,4 +65,39 @@ protected function postExecute(Message $message)
      * @return boolean
      */
     abstract protected function execute(Message $message);
+
+    /**
+     * Event fired when worker has processed message successfully.
+     *
+     * @return void
+     */
+    abstract protected function onSucceeded();
+
+    /**
+     * Event fired when worker has failed to process message.
+     *
+     * @return void
+     */
+    abstract protected function onFailed();
```

Comment:
Same reason with onSucceeded() method.

Reply:
Furthermore, we could break all current workers of developers after updating this PR.